libtest/signal.cc

```diff
@@ -117,6 +117,25 @@ bool SignalThread::unblock()
   return true;
 }
 
+std::string SignalThread::random_lock_name(std::string::size_type len = 10)
+{
+  static auto& chrs = "0123456789"
+    "abcdefghijklmnopqrstuvwxyz"
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+  thread_local static std::mt19937 rg{std::random_device{}()};
+  thread_local static std::uniform_int_distribution<std::string::size_type> pick(0, sizeof(chrs) - 2);
+
+  // convenient to named semaphore defination
+  // http://man7.org/linux/man-pages/man7/sem_overview.7.html
+  std::string s{"/"};
+  s.reserve(len--);
+
+  while(len--)
+    s += chrs[pick(rg)];
```

Comment:
Something tells me that the coding style for this project is to put braces around single statement loops like this. Please confirm.

Reply:
I'll add the braces.